compiler/rustc_lint/src/types.rs

```diff
@@ -1658,6 +1670,9 @@ impl<'a, 'tcx> ImproperCTypesVisitor<'a, 'tcx> {
             FfiResult::FfiUnsafe { ty, reason, help } => {
                 self.emit_ffi_unsafe_type_lint(ty, sp, reason, help);
             }
+            FfiResult::RecursionLimitReached => {
+                self.cx.tcx.dcx().emit_err(RecursionLimitReached { ty, span: sp });
```

Comment:
This shouldn't be a recursion limit error. It should probably be under the same FFI-unsafety warning but with adjusted wording like "couldn't determine if infinitely recursive type is FFI safe"